crates/polars-lazy/src/frame/mod.rs

```diff
@@ -730,18 +730,42 @@ impl LazyFrame {
     pub fn collect(self) -> PolarsResult<DataFrame> {
         #[cfg(feature = "new_streaming")]
         {
-            if self.opt_state.new_streaming {
-                let alp_plan = self.to_alp_optimized()?;
+            let slf = std::panic::AssertUnwindSafe(&self);
+            let run_on_new_streaming_engine = move || {
+                let alp_plan = slf.clone().to_alp_optimized()?;
                 let lp_top = alp_plan.lp_top;
-                let mut ir_arena = alp_plan.lp_arena;
-                let expr_arena = alp_plan.expr_arena;
+                let mut ir_arena = alp_plan.lp_arena.clone();
```

Comment:
Why do we need to clone here? Can't we just `take`?

Reply:
Because the engine might fail and then we have to fall back.